src/storage/src/table/batch_table/storage_table.rs

```diff
@@ -83,6 +85,13 @@ pub struct StorageTableInner<S: StateStore, SD: ValueRowSerde> {
 
     /// Row deserializer to deserialize the whole value in storage to a row.
     row_serde: Arc<SD>,
+    /// The `Deserializer` deserializes the columns specified in `value_output_indices`.
+    /// The `Deserializer` ensures there are no duplicate columns because `ColumnAwareSerde` disallows them.
+    /// The `ColumnMapping` restores all columns specified in `value_output_indices`.
+    partial_row_deserializer: Option<(
+        Arc<column_aware_row_encoding::Deserializer>,
+        Arc<ColumnMapping>,
```

Comment:
Why do we need both `row_serde` and `partial_row_deserializer`? Is there a case when we need to access the full row even though the `value_output_indices` is set to only ouputing a subset of columns in value?

Reply:
I kept both for potential usage in the future. 
I'll unify them.